enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -399,14 +325,18 @@ anyway.
 
 ### Risks and Mitigations
 
-While this proposal settles on installing via CVO, that decision has ramifications whose discussion are noted above.
-It is at least conceivable the need to revisit that decision may arise when implementation starts.
+**Risk:** Deploying via OLM and the cluster storage operator can lead to confusion
 
-Getting the flow and various stages from the CSI specification understood and correctly utilized so that we minimize
-the amount of times the Secret is read and stored on the local filesystem of each host/node is important.  We want
-to minimize local file system writes to when the Secret actually changes, and not when we provision the volume to
-a new Pod.
+*Mitigations:*
 
+- OLM version of the CSI driver operator will only be made available as a community operator
+- Users will be informed that this operator will need to be uninstalled prior to upgrading to the version of OpenShift that deploys this CSI driver via the OCP payload.
+
+**Risk:** CSI driver will not scale
+
+*Mitigations:*
+
+- Minimize local file system writes to when the Secret actually changes.
```

Comment:
Please include some scale tests. How many ProjectedResources  do you expect? How many namespaces / pods that use them do you expect?

Reply:
Yep we have scale testing as part of the GA / productization epic/stories I mentioned above.  I believe I can add that as part of the existing e2e's I have (albeit as optional at least initially).

But yes let's make noted of that here @adambkaplan

Reply:
I can see two forms of scale testing needed:

1. Number of total shares in the cluster - how many can a cluster handle?
2. Number of pods that can share a single share. RHEL entitlements come to mind - what if 1000 pods mounted the same simple content access cert? What happens if that SCA is rotated and 1000 pods consume it?

Reply:
My concerns also included
3. Number of namespaces to monitor for secrets/configmapas
4. Number of total secrets/configmaps as a result of the number of namespaces we watch

1. is something as well in my mind, but not as big a deal for me as 3 and 4 ... I think in practice the number of shares will be relatively smaller
2.  is something that we should pay due diligence on, but is the least of my concerns based on a conversation with @derekwaynecarr during a recent on top pillar arch call .... bottom line, this boils down the tmpfs mounts, of which clusters already have a bunch of since that is how secrets work ... he expressed confidence to me this would be "OK"